src/torch_constrained/constrained_optimizer.py

```diff
@@ -1,359 +1,289 @@
-import functools
 import inspect
 import warnings
-from typing import Type, Callable, Optional
+from functools import partial
+from typing import Type, List, Callable, Optional, Dict
 
 import torch
-import torch.nn
-
+from .multipliers import _SparseMultiplier, _DenseMultiplier
 
 class ConstrainedOptimizer(torch.optim.Optimizer):
     def __init__(
             self,
-            loss_optimizer: Type[torch.optim.Optimizer],
-            constraint_optimizer: Type[torch.optim.Optimizer],
-            lr_x,
-            lr_y,
+            primal_optim_class: Type[torch.optim.Optimizer],
+            primal_kwargs: Dict,
             primal_parameters,
-            augmented_lagrangian_coefficient=False,
+            dual_optim_class: Optional[Type[torch.optim.Optimizer]]=None,
+            dual_kwargs: Optional[Dict]=None,
+            ineq_init: Optional[List[torch.Tensor]]=None,
+            eq_init: Optional[List[torch.Tensor]]=None,
+            augmented_lagrangian_coeff=False,
             alternating=False,
-            shrinkage: Optional[Callable] = None,
-            dual_dtype=None,
-            primal_kwargs={},
-            dual_kwargs={},
-    ):
+            shrinkage: Optional[Callable]=None,
+            dual_reset=False,
```

Comment:
something like, dual_reset_on_feasibility based on the comment later, could make a more explicit name.
is it for ineq? for eq only?